collated/main/Jeremy.md

```diff
@@ -0,0 +1,1529 @@
+            Bloodtype bloodType = (!arePrefixesPresent(argMultimap, PREFIX_BLOODTYPE))
+                    ? new Bloodtype(NON_COMPULSORY_BLOODTYPE)
+                    : ParserUtil.parseBloodType(argMultimap.getValue(PREFIX_BLOODTYPE)).get();
+            Remark remark = (!arePrefixesPresent(argMultimap, PREFIX_REMARK))
+                    ? new Remark("") : ParserUtil.parseRemark(argMultimap.getValue(PREFIX_REMARK)).get();
+```
+###### \java\seedu\address\logic\parser\AddressBookParser.java
+``` java
+
+    /**
+     * Returns the correct list feature based on word after list
+     *
+     * @param arguments full user input arguments
+     * @return the command based on arguments provided
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    private Command listEvaluator(String arguments) throws ParseException {
+        String[] argSplit = arguments.trim().split(" ");
+        String firstArg = argSplit[0];
+        int firstArgLength = firstArg.length();
+        Command returnThisCommand;
+        switch (firstArg) {
+        case "":
+            returnThisCommand = new ListCommand();
+            break;
+        case "tag":
+            returnThisCommand = new ListByTagCommandParser().parse(arguments.substring(firstArgLength + 1));
+            break;
+        case "asc":
+        case "ascending":
+            returnThisCommand = (argSplit.length == 1) ? new ListAscendingNameCommand() : new ListFailureCommand();
+            break;
+        case "dsc":
+        case "descending":
+            returnThisCommand = (argSplit.length == 1) ? new ListDescendingNameCommand() : new ListFailureCommand();
+            break;
+        case "rev":
+        case "reverse":
+            returnThisCommand = (argSplit.length == 1) ? new ListReverseCommand() : new ListFailureCommand();
+            break;
+        default:
+            returnThisCommand = new ListFailureCommand();
+        }
+        return returnThisCommand;
+    }
+```
+###### \java\seedu\address\logic\parser\ListByTagCommandParser.java
+``` java
+
+/**
+ * Parses input arguments and creates a new ListByTagCommand object
+ */
+public class ListByTagCommandParser implements Parser<ListByTagCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns an FindCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ListByTagCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+
+        String[] tagKeyWords = trimmedArgs.split("\\s+");
+        List<String> evaluateList = Arrays.asList(tagKeyWords);
+
+        if (trimmedArgs.isEmpty() || invalidListTagArgs(evaluateList)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    ListByTagCommand.MESSAGE_USAGE));
+        }
+
+
```

Comment:
Use blank lines sparingly. It's good to have blank lines to separate logical code blocks, but too many blank lines will increase the height of the code, which increases the amount of scrolling.